lib/job-iteration/iteration.rb

```diff
@@ -266,18 +268,22 @@ def job_should_exit?
       JobIteration.interruption_adapter.call || (defined?(super) && super)
     end
 
-    def run_complete_callbacks?(completed)
-      # nil means that someone aborted the job but want to call the on_complete callback
-      if completed.nil?
-        completed = :finished
-      end
-
+    def handle_completed(completed)
       case completed
-      when :finished, true then true
-      # skip_complete_callbacks is returning from ThrottleEnumeratorand we do not want the on_complete callback to
-      # be executed
-      when false, :skip_complete_callbacks then false
+      when nil # someone aborted the job but want to call the on_complete callback
+        return true
+      when :completed
```

Comment:
This is a bit incompatible, technically we should still handle `:finished` and `true`. Someone may have had a `throw(:abort, true)` or `throw(:abort, :finished)` to stop and run the complete callbacks. But this was neither tested not documented.